### PR TITLE
Add Priority Scores

### PR DESCRIPTION
**Pull request by JasonMHasperhoven**
As part of https://github.com/orgs/prax-wallet/projects/1/views/7?pane=issue&itemId=82180074&issue=penumbra-zone%7Cweb%7C1829

### CODE REVIEW

File: npm/.changeset/cyan-cheetahs-carry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@penumbra-labs/registry': patch
```
Comment (grod220) [inline npm/.changeset/cyan-cheetahs-carry.md:2]: issue: this should be a minor bump given it is a forward-compatible new feature

File: tools/compiler/src/processor.rs [hunk 2]
```diff
@@ -124,6 +126,12 @@ pub fn transport_metadata_along_channel(
     // Without this, decoding will fail because the asset ID won't match.
     pb_metadata.penumbra_asset_id = None;
 
+    // Add priority score
+    pb_metadata.priority_score = priority_scores_by_base
+        .get(&pb_metadata.base)
+        .copied()
+        .unwrap_or(1);
```
Comment (grod220) [inline tools/compiler/src/processor.rs:133]: issue: if it is not in the HashMap, don't think we should add any score to the metadata object. Aka:


```rust
    // Add priority score if available
    if let Some(score) = priority_scores_by_base.get(&pb_metadata.base) {
        pb_metadata.priority_score = *score;
    }
```

File: tools/compiler/src/parser.rs [hunk 3]
```diff
@@ -27,6 +28,7 @@ pub struct ChainConfig {
     pub ibc_connections: Vec<IbcInput>,
     pub native_assets: Vec<Metadata>,
     pub canonical_numeraires: Vec<String>,
+    pub priority_scores_by_base: HashMap<String, u64>,
```
Comment (grod220) [inline tools/compiler/src/parser.rs:31]: suggestion: Can you alias the string? Aka:

```rust
type BaseDenom = String

pub priority_scores_by_base: HashMap<BaseDenom, u64>,
```

File: input/chains/penumbra-1.json [hunk 4]
```diff
@@ -101,7 +101,84 @@
...
+    "transfer/channel-0/uatom": 18,
+    "transfer/channel-4/factory/osmo12lnwf54yd30p6amzaged2atln8k0l32n7ncxf04ctg7u7ymnsy7qkqgsw4/alloyed/allTON": 18,
+    "transfer/channel-4/factory/osmo1g8qypve6l95xmhgc0fddaecerffymsl7kn9muw/sqbtc": 1,
+    "transfer/channel-4/factory/osmo1xqw2sl9zk8a6pch0csaw78n4swg5ws8t62wc5qta4gnjxfqg6v2qcs243k/stuibcx": 1,
+    "transfer/channel-4/factory/osmo1dywfmhyc8y0wga7qpzej0x0mgwqg25fj4eccp494w8yafzdpgamsx9ryyv/fGECK": 1,
+    "transfer/channel-4/factory/osmo1g8qypve6l95xmhgc0fddaecerffymsl7kn9muw/sqatom": 17,
+    "transfer/channel-4/factory/osmo1q77cw0mmlluxu0wr29fcdd0tdnh78gzhkvhe4n6ulal9qvrtu43qtd0nh8/crazyhorse": 1,
+    "transfer/channel-4/uosmo": 14,
+    "transfer/channel-4/factory/osmo10n8rv8npx870l69248hnp6djy6pll2yuzzn9x8/BADKID": 1,
+    "upenumbra": 99,
```
Comment (grod220) [inline input/chains/penumbra-1.json:128]: suggestion: can you move this to the top? Actually, can we rank these be priority scores descending?

File: input/chains/penumbra-1.json [hunk 5]
```diff
@@ -101,7 +101,84 @@
...
-  ]
+  "canonicalNumeraires": ["transfer/channel-2/uusdc"],
+  "priorityScoresByBase": {
+    "transfer/channel-4/factory/osmo1q77cw0mmlluxu0wr29fcdd0tdnh78gzhkvhe4n6ulal9qvrtu43qtd0nh8/COOK": 1,
+    "transfer/channel-4/factory/osmo1dywfmhyc8y0wga7qpzej0x0mgwqg25fj4eccp494w8yafzdpgamsx9ryyv/fSLOTH": 1,
+    "transfer/channel-4/factory/osmo1vdvnznwg597qngrq9mnfcfk0am9jdc9y446jewhcqdreqz4r75xq5j5zvy/ymos": 1,
+    "transfer/channel-4/factory/osmo1pfyxruwvtwk00y8z06dh2lqjdj82ldvy74wzm3/WOSMO": 14,
+    "transfer/channel-4/factory/osmo19hdqma2mj0vnmgcxag6ytswjnr8a3y07q7e70p/wLIBRA": 1,
+    "transfer/channel-4/factory/osmo14mafhhp337yjj2aujplawz0tks6jd2lel4hkwz4agyzhvvztzaqsqzjq8x/alloyed/allTRX": 18,
+    "transfer/channel-4/factory/osmo1q77cw0mmlluxu0wr29fcdd0tdnh78gzhkvhe4n6ulal9qvrtu43qtd0nh8/cac": 1,
```
Comment (grod220) [inline input/chains/penumbra-1.json:112]: question: Is there a reason why we would add a priority score of 1 to any? Feels like we can drop it entirely if it's that low

File: input/chains/penumbra-1.json (hunk 4 above)
Comment (grod220) [reply input/chains/penumbra-1.json:128]: also, I'd make this 999999999999

**Review by grod220 — APPROVED** [review]
Great job on your first Rust PR! 🎊

File: input/chains/penumbra-1.json (hunk 4 above)
Comment (hdevalence) [reply input/chains/penumbra-1.json:128]: yeah, I think a good approach is:

- only specify overrides (if unspecified, will use a default score of 0)
- specify assets top to bottom
- leave spaces in the numbering (can put UM as 1000000, other stuff at 100000)

**Pull request merged**